### ISSUE TITLE
Media transparent background flickers on high refresh rate displays

### ISSUE THREAD

**Issue by costentz** [telegramdesktop/tdesktop#25586]
### Steps to reproduce

1. Go to any chat on telegram desktop for Windows with a High refresh rate display (120hz in my case);
2. Open any media and reduce its size to have the grey transparent background.

### Expected behaviour

There should be no flickering.

### Actual behaviour

The grey transparent background have brightness flickering, which is painful for the eyes. It's the same thing that happens in games wihen the refresh rate go below the freesync ranges, but it should not be the case since it's a normal app.
It does not happen if I set my refresh rate at 60hz.

### Operating system

Windows 10

### Version of Telegram Desktop

4.4.1

### Installation source

Static binary from official website

**Comment by Aokromes** [comment]
if it's amd bug then maybe you need to report to amd.

**Comment by soredake** [comment]
> if it's amd bug then maybe you need to report to amd.

It's a monitor problem, cannot really be fixed unless you change your monitor to one that don't have this problem.

**Comment by Aokromes** [comment]
> > if it's amd bug then maybe you need to report to amd.
> 
> It's a monitor problem, cannot really be fixed unless you change your monitor to one that don't have this problem.

yes, but adding it to disable freesync thing maybe must be done by amd.

**Comment by igoorboob** [comment]
I can confirm that i have the same problem.
My specs: Xiaomi Mi 2K gaming monitor (165hz), AMD rx6600xt.
@costantino2000 @soredake can you share your monitor and gpu model?

Apparently I managed to fix the problem by setting the ANGLE graphics backend -> disabled (settings->advanced)

**Comment by soredake** [comment]
@igoorboob HP X24ih, amd rx 560

**Comment by igoorboob** [comment]
> Got the same issue on Gigabyte M34WQ and 7900XTX, no way I'm enabling a software renderer - ANGLE - to address this.

ANGLE was Auto by default (at least that's how it was for me). Try setting it to disabled and restart Telegram desktop app

**Comment by Klaster1** [comment]
@igoorboob thanks for the advice, I tried all the ANGLE backend options:
1. D3D11: no full screen flickering, although video embeds in posts do flicker.
2. D3D9: green tint in full screen videos, no flicker.
3. D3D11on12: full screen lightbox is invisible, can't watch fullsized video embeds.
4. OpenGL: nigh unusable, everything's hidden when a full screen lightbox closes, resets HDR.
5. Disabled: default, other apps flickers (what topic started with).

I'll see how D3D11 BE fares.

**Comment by igoorboob** [comment]
@Aokromes 
Probably the problem is really mostly on the AMD side, but maybe Telegram Desktop developers can try to contact the AMD guys? Perhaps they will put Telegram Desktop in the exceptions for FreeSync at the driver level?
The problem really ruins the experience of interacting with Telegram Desktop. AMD GPUs certainly have less distribution than Nvidia, but it still affects thousands of users.

**Comment by MadHundred** [comment]
still have this error till latest driver AMD driver : `23.9.3`
The problem can be solved by disabling the `AMD FreeSync Premium` on AMD Software, because this problem will happen with Microsoft Excel too.
Go to **AMD Software (Adrenalin)** , then go to **Gaming** , in **Display** tab , **Uncheck the AMD FreeSync**

**Comment by Klaster1** [comment]
The issue still happens for me, same as for the people above.

**Comment by herhor67** [comment]
Whenever I open any media, the entire screen flickers until I close the app entirely.
None of the ANGLE settings change anything.
Radeon 5700 stock with Q27G2WG4

**Comment by Aphotic** [comment]
> I just reported the bug with the Bug Report Tool, hopefully they will look into it.

![image](https://github.com/telegramdesktop/tdesktop/assets/30921064/513b1b7c-5539-4c7d-87f1-8074d40440c4)
still have this issue on 24.6.1v

**Comment by Aphotic** [comment]
i might have found a workaround/fix:

using CRU-1.5.1(Custom Resolution Utility) by ToastyX
to set my min FPS to a higher than normal
![image](https://github.com/telegramdesktop/tdesktop/assets/30921064/5a77c714-d110-4faf-9c24-54cd822567a9)

test and report back until Telegram/AMD fixes this

**Comment by Aphotic** [comment]
another workaround i found is to add telegram(any app that has Free Sync issues)to AMD Adrenaline:
![image](https://github.com/telegramdesktop/tdesktop/assets/30921064/91f0a102-a57a-4a68-9836-70b0ccaf0675)
and disable the Free Sync on that individual program:
![image](https://github.com/telegramdesktop/tdesktop/assets/30921064/a7a0835f-bed9-4826-a5b5-5091cbd345b3)

**Comment by costantino2000** [comment]
I tried the CRU method but it did not work, the Adrenaline method is the one I've been using for a long time and it works fine.

**Comment by Aphotic** [comment]
> the Adrenaline method works fine.
hmm for me none worked at the end, what exactly are you doing give us your monoitor specs and how it flickers for you . for me when i go watch anything full-screen video/picture, when i come our of fullscreen i still get the flickers until i switch apps for a second.

**Comment by Aphotic** [comment]
> the Adrenaline method works fine.

hmm for me none worked at the end, what exactly are you doing give us your monoitor specs and how it flickers for you . for me when i go watch anything full-screen video/picture, when i come our of fullscreen i still get the flickers until i switch apps for a second.

**Comment by costantino2000** [comment]
> what exactly are you doing give us your monoitor specs and how it flickers for you 

I have a 160hz monitor and I'm on the 23.11.2 version of AMD's drivers; Telegram only flickers for a second when scrolling through the media (I guess it's caused by some sort of frame drop when it loads the next media).
I have been using CRU for quite some time, I tried changing the minimum refresh rate from 48 to 85 like you did but it started making the screen flicker all the time, so I reversed it.
As I said, I fixed the problem a long time ago adding Telegram to the games section in Adrenaline and disabling the Freesync for it, it also worked when the flickering always happened while viewing any kind of media last year.

**Comment by soredake** [comment]
Not stale

**Comment by Fixer-007** [comment]
I have the same issue on 240Hz monitor on win 10 with freesync premium enabled, rx6600, amd driver 25.10.2, latest telegram client. 
Workaround by Aphotic actually solved the issue.

**Comment by Aokromes** [comment]
https://github.com/telegramdesktop/tdesktop/issues/25586#issuecomment-1482987119
